x-pack/plugins/endpoint/common/generate_data.test.ts

```diff
@@ -0,0 +1,140 @@
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { EndpointDocGenerator, Event } from './generate_data';
+
+interface Node {
+  events: Event[];
+  children: Node[];
+  parent_entity_id?: string;
+}
+
+describe('data generator', () => {
+  let generator: EndpointDocGenerator;
+  beforeEach(() => {
+    generator = new EndpointDocGenerator();
+  });
+
+  it('creates endpoint metadata documents', () => {
+    const timestamp = new Date().getTime();
+    const metadata = generator.generateEndpointMetadata(timestamp);
+    expect(metadata['@timestamp']).toEqual(timestamp);
+    expect(metadata.event.created).toEqual(timestamp);
+    expect(metadata.endpoint).not.toBeNull();
+    expect(metadata.agent).not.toBeNull();
+    expect(metadata.host).not.toBeNull();
+  });
+
+  it('creates alert event documents', () => {
+    const timestamp = new Date().getTime();
+    const alert = generator.generateAlert(timestamp);
+    expect(alert['@timestamp']).toEqual(timestamp);
+    expect(alert.event.action).not.toBeNull();
+    expect(alert.endpoint).not.toBeNull();
+    expect(alert.agent).not.toBeNull();
+    expect(alert.host).not.toBeNull();
+    expect(alert.process.entity_id).not.toBeNull();
+  });
+
+  it('creates process event documents', () => {
+    const timestamp = new Date().getTime();
+    const processEvent = generator.generateEvent({ timestamp });
+    expect(processEvent['@timestamp']).toEqual(timestamp);
+    expect(processEvent.event.category).toEqual('process');
+    expect(processEvent.event.kind).toEqual('event');
+    expect(processEvent.event.type).toEqual('creation');
+    expect(processEvent.agent).not.toBeNull();
+    expect(processEvent.host).not.toBeNull();
+    expect(processEvent.process.entity_id).not.toBeNull();
+  });
+
+  it('creates other event documents', () => {
+    const timestamp = new Date().getTime();
+    const processEvent = generator.generateEvent({ timestamp, eventCategory: 'dns' });
+    expect(processEvent['@timestamp']).toEqual(timestamp);
+    expect(processEvent.event.category).toEqual('dns');
+    expect(processEvent.event.kind).toEqual('event');
+    expect(processEvent.event.type).toEqual('creation');
+    expect(processEvent.agent).not.toBeNull();
+    expect(processEvent.host).not.toBeNull();
+    expect(processEvent.process.entity_id).not.toBeNull();
+  });
+
+  it('creates alert ancestor tree', () => {
+    const events = generator.generateAlertEventAncestry(3);
+    for (let i = 1; i < events.length - 1; i++) {
+      expect(events[i].process.parent?.entity_id).toEqual(events[i - 1].process.entity_id);
+      expect(events[i].event.kind).toEqual('event');
+      expect(events[i].event.category).toEqual('process');
+    }
+    // The alert should be last and have the same entity_id as the previous process event
+    expect(events[events.length - 1].process.entity_id).toEqual(
```

Comment:
it would probably be better to break some of these assertions into separate `it`'s. You could have something like:
```
describe('when creating an alert ancestor tree', () => {
  it('the alert should be last', () => {}
}
```